### PR TITLE
Fix unauthorized errors flow

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Fix unauthorized errors flow.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-3060

### Is it tested? How?

1. Install CHE on OpenShift.
2. Set cluster accessTokenInactivityTimeout = 5 minutes:
```oc patch OAuth/cluster -p '{ "spec": { "tokenConfig": { "accessTokenInactivityTimeout": "5m" }}}' --type merge```
3. Open CHE Dashboard page in Google Chrome.
4. Leave it along about 20 minutes.
5. Refresh CHE Dashboard page.
6. CHE Dashboard should be redirected to the login page.

You can tested it with https://eclipse-che.apps.cluster-7vz9l.7vz9l.sandbox688.opentlc.com/dashboard/

### CODE REVIEW

File: packages/dashboard-frontend/src/services/workspace-client/index.ts [hunk 1]
```diff
@@ -32,54 +31,36 @@ export abstract class WorkspaceClient {
...
-          (isUnauthorized(response) || isForbidden(response))
-        ) {
-          await this.deauthorize();
+        const isApi = this.checkPathPrefix(response, '/api/');
+        if (isApi) {
+          if (isUnauthorized(response)) {
+            signIn();
+          } else if (isForbidden(response)) {
+            signOut();
+          }
```
Comment (ibuziuk) [inline packages/dashboard-frontend/src/services/workspace-client/index.ts:40]: @olexii4 got a bit lost why do we sign in on 401 and sign out on 403?

Comment (olexii4) [reply packages/dashboard-frontend/src/services/workspace-client/index.ts]: It was a part of the old implementation. I have fixed it.

**Review by akurinnoy — APPROVED** [review]
LGTM

**Review by ibuziuk — APPROVED** [review]

**Comment by nickboldt** [comment]
building in https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/CRW_CI/job/dashboard_3.x/52/

I'm going to move this to the 7.50 milestone since 7.49 was a week ago on June 8

**Pull request merged**